src/support/unique_deferring_queue.h

```diff
@@ -57,6 +57,32 @@ template<typename T> struct UniqueDeferredQueue {
       // skip this one, keep going
     }
   }
+
+  void clear() {
+    while (!data.empty()) {
+      data.pop();
+    }
```

Comment:
How about
```suggestion
    std::queue<int> empty;
    std::swap(data, empty);
```

Reply:
That is nicer, thanks. Weird std::queue doesn't have a `clear()` method...